interactions/api/gateway.py

```diff
@@ -1,430 +1,321 @@
+        "ready",
     )
 
     def __init__(
         self,
+        token: str,
         intents: Intents,
-        session_id: Optional[int] = None,
-        sequence: Optional[int] = None,
+        session_id: Optional[int] = MISSING,
+        sequence: Optional[int] = MISSING,
     ) -> None:
         """
-        :param intents: The intents used for identifying the connection.
+        :param token: The token of the application for connecting to the Gateway.
+        :type token: str
+        :param intents: The Gateway intents of the application for event dispatch.
         :type intents: Intents
-        :param session_id?: The session ID if you're trying to resume a connection. Defaults to ``None``.
+        :param session_id?: The ID of the session if trying to reconnect. Defaults to ``None``.
         :type session_id: Optional[int]
-        :param sequence?: The sequence if you're trying to resume a connection. Defaults to ``None``.
+        :param sequence?: The identifier sequence if trying to reconnect. Defaults to ``None``.
         :type sequence: Optional[int]
         """
-        self.intents = intents
-        self.loop = get_event_loop()
-        self.dispatch = Listener()
-        self.session = None
-        self.session_id = session_id
-        self.sequence = sequence
-
-        self.keep_alive = None
-        self.closed = False
-        self.http = None
-        self.options: dict = {
+        try:
+            self._loop = get_event_loop() if version_info < (3, 10) else get_running_loop()
+        except RuntimeError:
+            self._loop = new_event_loop()
+        self._dispatch = Listener()
+        self._http = HTTPClient(token)
+        self._client = None
+        self._closed = False
+        self._options = {
             "max_msg_size": 1024 ** 2,
             "timeout": 60,
             "autoclose": False,
             "compress": 0,
         }
-
-    async def recv(self) -> Optional[Any]:
-        """Receives packets sent from the gateway."""
-        packet = await self.session.receive()
-        return (
-            loads(packet.data)
-            if packet and isinstance(packet.data, (bytearray, bytes, memoryview, str))
-            else None
+        self._intents = intents
+        self.__heartbeater: _Heartbeat = _Heartbeat(
+            loop=self._loop if version_info < (3, 10) else None
         )
+        self.__shard = None
+        self.__presence = None
+        self.__task = None
+        self.session_id = None if session_id is MISSING else session_id
+        self.sequence = None if sequence is MISSING else sequence
+        self.ready = None
+
+    @property
+    async def __heartbeat_manager(self) -> None:
+        """Manages the heartbeat loop."""
+        while True:
+            if self.__heartbeater.event.is_set():
+                await self.__heartbeat_packet
+                self.__heartbeater.event.clear()
+                await sleep(self.__heartbeater.delay / 1000)
```

Comment:
You should use `asyncio.wait` so that you can wait for the first to complete (event or sleep).

Reply:
Ah update on this, I was a bit confused by `self.__heartbeater.event`. I assumed it was meant to close the heartbeater, so this change is probably not what is appropriate.

Reply:
We refer to `__task` as our attempt to start a new task in closing the event fyi